tools/storybook/src/addons/docs/index.tsx

```diff
@@ -0,0 +1,198 @@
+import { API, useParameter } from "@storybook/manager-api";
+import type { NamedType, NodeType, ObjectType } from "@player-tools/xlr";
+import { WithTooltip, SyntaxHighlighter, Link } from "@storybook/components";
+import ts from "typescript";
+import React from "react";
+import { isPrimitiveTypeNode, TSWriter } from "./ts-convert";
+import { useDarkMode } from "../useDarkMode";
+
+interface DocsPanelProps {
+  /** if the panel is actively shown to the user */
+  active: boolean;
+
+  /** Storybook API */
+  api: API;
+}
+
+function getKeyName(node: NodeType): string {
+  if (node.type === "ref") {
+    return node.ref;
+  }
+
+  if (node.type === "or") {
+    return node.or.map((subnode) => getKeyName(subnode)).join(" | ");
+  }
+
+  if (node.type === "and") {
+    return node.and.map((subnode) => getKeyName(subnode)).join(" & ");
+  }
+
+  if (node.type === "array") {
+    return `Array<${getKeyName(node.elementType)}>`;
+  }
+
+  if (node.type === "record") {
+    return `Record<${getKeyName(node.keyType)}, ${getKeyName(node.valueType)}>`;
+  }
+
+  if (isPrimitiveTypeNode(node) && node.type !== "null") {
+    return node.type;
+  }
+
+  if (node.type === "object" && node.name) {
+    return node.name;
+  }
+
+  return node.type;
+}
+
+function XLRObjectDocs(props: {
```

Comment:
Moving this here would be great. We'll probably need to figure out how we want to go about deprecating the webpack xlr plugin

Reply:
Is there anything that would stop us from just not publishing new versions? It wouldn't work with the updated storybook plugin updates for v7 anyway

Reply:
The usage instructions were already updated on the storybook page: https://github.com/player-ui/player/pull/449/files#diff-bb7fbe3850070f90366f286656bbc59ebfb8a8734d46fd8796cd1738f99d40e1R71